src/daft-recordbatch/src/ops/window.rs

```diff
@@ -127,19 +130,102 @@ impl RecordBatch {
             }
         };
 
-        self.window_agg_rows_between(
-            agg_expr,
-            &name,
-            dtype,
-            start_boundary,
-            end_boundary,
-            min_periods,
-            total_rows,
-        )
+        match agg_expr {
+            AggExpr::Sum(_)
+            | AggExpr::Count(..)
+            | AggExpr::Mean(_)
+            | AggExpr::Min(_)
+            | AggExpr::Max(_)
+            | AggExpr::CountDistinct(_) => self.window_agg_rows_optimized(
+                agg_expr,
+                &name,
+                start_boundary,
+                end_boundary,
+                min_periods,
+                total_rows,
+            ),
+            _ => self.window_agg_rows(
+                agg_expr,
+                &name,
+                dtype,
+                start_boundary,
+                end_boundary,
+                min_periods,
+                total_rows,
+            ),
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn window_agg_rows_optimized(
```

Comment:
This is not a very descriptive name. There's other optimized ways to compute window as well. Let's give it a better name + also add some doc strings to describe what it does